### DISCUSSION TITLE
Pimcore Admin UI shows object data from versions instead of object_* tables (object.o_versionCount < versions.versionCount Problem?)

### DISCUSSION THREAD

**Discussion by chm0815** [pimcore/pimcore#16022] [Q&A / Support] [answered]
When opening some specific objects in the Admin UI, Pimocore displays the object data coming from the versions (with highest versionCount). This happens because of [getLatestVersion ](https://github.com/pimcore/pimcore/blob/0fbdbff0e54fe65af2225f3f4e30062c11e9b272/models/Element/Traits/VersionDaoTrait.php#L32)

My object has a smaller o_versionCount than versionCount in the versions table.
For me the question is now, why is the o_versionCount smaller than in the versions table or if there was a problem updating the objects.o_versionCount. 

I found several non-autoSave, non-draft objects with a lower o_versionCount than the value stored in the versions table with the following query? 

```
select o.o_id,v.id,o.o_path,o.o_versionCount,v.versionCount,FROM_UNIXTIME(o.o_modificationDate),FROM_UNIXTIME(v.`date`)  
from objects o, versions v
 where o.o_id = v.cid and o.o_versionCount < v.versionCount and v.autoSave = 0 and ctype='object';
```

Could someone explain this phenomenon or might there be a bug?

Pimcore Version: 10.3.7

**Comment by chm0815** [comment]
It seems, there is a serious problem caused by a **race condition**. 
I found one way to reproduce the problem (**objects.o_versionCount < versions.versionCount**) for non-draft objects.

Imagine you have a command that saves objects and has **disabled version creation**.
Eg.
```php
Version::disable();
$obj = DataObject::getById(1234);
// sleep(100) or set a breakpoint. In the meantime, save the same object with enabled versions (eg via the Admin UI)
$obj->save();

```

The problem now is that the object saved with the code above has now a smaller o_versionCount than the newest version in the versions table, because the o_versionCount is only updated when the version creation is enabled [here].https://github.com/pimcore/pimcore/blob/b147e218e55da7cb2bb3023a6b499cda7021f6f9/models/Element/AbstractElement.php#L296.

The object edit UI will potentially display different data than what is stored in the object_* tables what is very misleading for the user, because of [getLatestVersion](https://github.com/pimcore/pimcore/blob/0fbdbff0e54fe65af2225f3f4e30062c11e9b272/models/Element/Traits/VersionDaoTrait.php#L32).  

There was already a similar issue here: https://github.com/pimcore/pimcore/issues/3523

**Reply by dvesh3** [reply]
@chm0815 please create an issue with bug label. thanks!

**Reply by chm0815** [reply, marked as answer]
Issue created here: https://github.com/pimcore/pimcore/issues/16039
